bal_addresses/queries.py

```diff
@@ -0,0 +1,155 @@
+from bal_addresses import AddrBook
+from .errors import GraphQLRequestError
+import requests
+from collections import defaultdict
+
+NO_BALANCER_SUBGRAPH = []
+NO_GAUGE_SUBGRAPH = ["bsc", "kovan", "fantom", "rinkeby"]
+NO_AURA_SUBGRAPH = ["avalanche"]
+
+
+class GraphEndpoints:
+    balancer = defaultdict(lambda: None)
+
+    gauges = defaultdict(lambda: None)
+    aura = defaultdict(lambda: None)
+    blocks = {"mainnet": "https://api.thegraph.com/subgraphs/name/blocklytics/ethereum-blocks",
+        "arbitrum": "https://api.thegraph.com/subgraphs/name/ianlapham/arbitrum-one-blocks",
+        "polygon": "https://api.thegraph.com/subgraphs/name/ianlapham/polygon-blocks",
+        "base": "https://api.studio.thegraph.com/query/48427/bleu-base-blocks/version/latest",
+        "gnosis": "https://api.thegraph.com/subgraphs/name/rebase-agency/gnosis-chain-blocks",
+        "avalanche": "https://api.thegraph.com/subgraphs/name/iliaazhel/avalanche-blocks",
+        "zkevm": "https://api.studio.thegraph.com/query/48427/bleu-polygon-zkevm-blocks/version/latest", }
+    blocks = defaultdict(lambda: None, blocks)
+
+    for chain in AddrBook.chain_ids_by_name.keys():
+        ## Mainnet often has a different URL string format
+        if chain == "mainnet":
+            balancer[chain] = "https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2"
+            gauges[chain] = "https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-gauges"
+            aura[chain] = "https://graph.aura.finance/subgraphs/name/aura/aura-mainnet-v2-1"
+        ## This handles other chains which tend to follow a pattern, with some chains sometimes missing depending on the subgraph
+        else:
+            ## balancer
+            if chain not in NO_BALANCER_SUBGRAPH:
+                balancer[chain] = f"https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-{chain}-v2"
```

Comment:
i dont think it is this easy, there quite some exceptions to this pattern. that is why i wrote `utils.get_subgraph_url`, which fetches the correct url straight from the production ui.

probably best to leverage the logic of that function, or maybe even move it here?

Reply:
Yeah it's not. It is in many cases but the naming changes. 

My thought was jto just collect whatever we need to collect here to try to get all the endpoints. Be that a static list or some pattern matching with then static overrides or something.  I think if you can figure out how to move it here that would be great, but there are a number of other subgraphs such as the aura subgraph, beefy, blocks, etc.

Any work you can do to make the lists of endpoints per subgraph per chain more complete/functional is gud.

Reply:
I am not a fan of fetching the subgraph endpoints from the PROD UI. We should rather strive for a central registry managed by the data team.